### ISSUE TITLE
systemCapabilities null error

### ISSUE THREAD

**Issue by lintunen** [Fernfrost/vsts-task-agent-capabilities#6]
Receiving the error:

```
2019-01-24T15:28:44.2507280Z ##[section]Starting: Agent Capabilities
2019-01-24T15:28:44.2511540Z ==============================================================================
2019-01-24T15:28:44.2511670Z Task         : Agent Capabilities
2019-01-24T15:28:44.2511790Z Description  : Exposes agent capabilities as variables in your pipeline. Use them just like any other variable in subsequent tasks.
2019-01-24T15:28:44.2511920Z Version      : 1.0.5
2019-01-24T15:28:44.2512010Z Author       : Fernfrost
2019-01-24T15:28:44.2512130Z Help         : Having Problems? Please [create an issue on our Github](https://github.com/Fernfrost/vsts-task-agent-capabilities/issues) and we will try to help you.
2019-01-24T15:28:44.2512550Z ==============================================================================
2019-01-24T15:28:45.5238890Z Agent capability variables and values. Format: variable=value
2019-01-24T15:28:45.5239600Z 
2019-01-24T15:28:45.5321810Z ##[error]Cannot read property 'systemCapabilities' of null
2019-01-24T15:28:45.5380660Z ##[section]Finishing: Agent Capabilities
```

Tried creating a new PAT but the problem still persists. Funnily enough the task is working on other projects. Using the same agent as other projects.

**Comment by lintunen** [comment]
I tried reusing and creating a new PAT and it made no difference. The agent was online, I tested it with another project to make sure the PAT was working.

The pipeline is a YAML build pipeline copied from another project. But if I have to re-add the task each time to get it to work (and by re-adding in this case I assume you mean just deleting the "-task" from the YAML and saving/running and adding it again) then this means I have to re-think using this task template.

**Comment by Fernfrost** [comment]
You shouldnt have to delete and re-add the task every time. Just trying to exclude all possible configuration errors.

Could you please run the build with debug logging and post the output? Hopefully that will shed some light on why it doesnt work for that specific project.

**Comment by Fernfrost** [comment]
I havnt been able to reproduce this.

Have you had any luck resolving the issue?

**Comment by aee11** [comment]
I also encountered this problem in a YAML pipeline with the following pool configuration:

```
pool:
  name: MyCustomPool
```

I noticed that the REST API response from "Get Build" in the failing build didn't return the `poolId` for `MyCustomPool`, rather it returned `Hosted Ubuntu 1604`'s `poolId`. So then I checked the YAML pipeline in the build definition UI and the `Agent Pool` parameter (under the YAML tab) was indeed set to `Hosted Ubuntu 1604`. I changed that to `MyCustomPool` and then the task ran successfully.

I'd say this is a bug in Azure DevOps. The UI for the failed build says the job was run on `MyCustomPool` but the REST response says something else.
